gulpfile.js/tasks/changelog.js

```diff
@@ -0,0 +1,70 @@
+'use strict'
+
+var gulp = require('gulp')
+var gutil = require('gulp-util')
+var proc = require('child_process')
+
+var runCommand = function (cmd) {
+  return new Promise(function (resolve, reject) {
+    proc.exec(cmd, function (err, stdout, stderr) {
+      if (err) {
+        reject(err)
+      } else if (stderr) {
+        reject(stderr)
+      } else {
+        resolve(stdout)
+      }
+    })
+  })
+}
+
+var getCurrentCommit = function (commit) {
+  return new Promise(function (resolve, reject) {
+    if (commit) {
+      resolve(commit)
+    } else {
+      var cmd = 'git rev-parse HEAD'
+      runCommand(cmd)
+        .then(function (commit) {
+          resolve(commit)
+        })
+    }
+  })
+}
+
+var getChangedFiles = function (commit) {
+  if (!commit) {
+    return Promise.reject(new Error('No commit given'))
+  }
+
+  var cmd = 'git diff --name-only master ' + commit
+  return runCommand(cmd)
+}
+
+var checkForChangelog = function (files) {
+  if (!files.includes('CHANGELOG.md')) {
```

Comment:
Yeah, there _is_ a mix of styles. I just got bored of writing the boilerplate of `return new` when hammering this out and then forgot to go back and standardise on one. Oops

I disagree with it being "a shorthand too far" though. If anything these returns shouldn't even be `Promise`s (as returning a value from a `then` callback is fine) but that was blowing up on Travis for some reason so "Promisify all the things" was the quickest/easiest way around it 😕

I'll go back and investigate further. Happy to pair on it if you want?